src/System.Net.Sockets/src/System/Net/Sockets/BufferList.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Net.Sockets
```

Comment:
As mentioned, this file is new: it contains an internal type I used for the Linux async engine to support polymorphism over `IList<ArraySegment<byte>>` and `BufferOffsetSize[]`.